llvm/lib/Transforms/IPO/Attributor.cpp

```diff
@@ -259,11 +259,16 @@ AA::getInitialValueForObj(Attributor &A, const AbstractAttribute &QueryingAA,
     if (!Initializer)
       return nullptr;
   } else {
-    if (!GV->hasLocalLinkage() &&
-        (GV->isInterposable() || !(GV->isConstant() && GV->hasInitializer())))
-      return nullptr;
-    if (!GV->hasInitializer())
-      return UndefValue::get(&Ty);
+    if (GV->hasLocalLinkage()) {
+      // uninitialized local variable
```

Comment:
nit: I'd not call them local variable as they are global variables with local linkage.